halo2_proofs/src/dev.rs

```diff
@@ -1130,4 +1220,654 @@ mod tests {
             }])
         );
     }
+
+    #[cfg(test)]
+    mod dynamic_lookups {
+        use crate::plonk::{DynamicTable, DynamicTableMap};
+
+        use super::*;
+
+        const K: u32 = 7;
+
+        #[derive(Clone)]
+        struct DynLookupCircuitConfig {
+            q: Selector,
+            a: Column<Advice>,
+            // Also contains zero to use as default
+            table_vals: Column<Advice>,
+            // A table containing table_vals.
+            table: DynamicTable,
+        }
+
+        impl DynLookupCircuitConfig {
+            fn assign_lookups(
+                &self,
+                layouter: &mut impl Layouter<Fp>,
+                lookup: impl Iterator<Item = usize> + Clone,
+            ) -> Result<(), Error> {
+                layouter.assign_region(
+                    || "lookups",
+                    |mut region| {
+                        // Enable the lookup on rows
+                        for (offset, val) in lookup.clone().enumerate() {
+                            self.q.enable(&mut region, offset)?;
+
+                            region.assign_advice(
+                                || "",
+                                self.a,
+                                offset,
+                                || Value::known(Fp::from(val as u64)),
+                            )?;
+                        }
+
+                        Ok(())
+                    },
+                )
+            }
+        }
+
+        /// Just like dynamic_lookup, but breaks the table into single row regions.
```

Comment:
I don't see how we break the table into single row regions. Am I missing anything?